Разметка всех страниц

OlgaNikl opened the pull request:
---
:mortar_board: [Разметка всех страниц](https://up.htmlacademy.ru/htmlcss/34/user/2067969/tasks/4)

OlgaNikl committed:
Update index.html

Размечает header, добавляет логотип, размечает навигацию

OlgaNikl committed:
Update index.html

Добавляет список пунктов навигации по сайту

OlgaNikl committed:
Update index.html

Исправляет скобку

OlgaNikl committed:
Update index.html

Добавляет список пользовательской навигации

OlgaNikl committed:
Update index.html

Размечает секции и заголовки основной части.

OlgaNikl committed:
Update index.html

Добавляет секцию приветствие

OlgaNikl committed:
Update index.html

добавляет комментарий

OlgaNikl committed:
Update index.html

добавляет список преимуществ

OlgaNikl committed:
Update index.html

добавляет комментарий

OlgaNikl committed:
Update index.html

добавляет список доп преимуществ  в список преимуществ

OlgaNikl committed:
Update index.html

добавляет изображение городка

OlgaNikl committed:
Update index.html

добавляет секцию поиск гостиницы

OlgaNikl committed:
Update index.html

размечает footer, добавляет секцию о подписке

OlgaNikl committed:
Update index.html

добавляет секцию с контактами

OlgaNikl committed:
Update index.html

Добавляет секцию с лого HTML академии

OlgaNikl committed:
Update index.html

Добавляет классы изображениям

talequale commented on index.html:
```diff
@@ -5,8 +5,151 @@
+        </ul>
+        <ul class="navigation-list navigation-user">
+          <li class="navigation-item">
+            <a class="navigation-link" href="#">
```
ссылки в этом списке совсем не похожи на ссылки в первом списке, лучше не делать их одним классом. в целом общего у двух списков мало и от попытки сделать их универсальными мы проиграем больше, чем выиграем. в барбершопе такое допустимо, потому что в обоих списках одинаковые текстовые характеристики, и айтемы и ссылки в этих списках можно считать разновидностями одного и того же компонента: у нас есть ссылка, она может состоять из текста и иконки, а может состоять либо только из текста, либо только из иконки. то есть в барбершопе в двух списках действительно присутствует универсальность, а в седоне нет. поэтому второму списку и его элементам лучше сразу дать другие классы. например, так:

```
<ul class="user-list">
    <li class="user-list-item">
        <a class="user-list-link" href="">иконка лупы</a>
    </li>
    <li class="user-list-item">
        <a class="user-list-link" href="">иконка сердечка</a>
    </li>
    <li class="user-list-item">
        <a class="user-list-button" href="">Хочу сюда!</a>
    </li>
</ul>
```

talequale commented on index.html:
```diff
@@ -5,8 +5,151 @@
+          </li>
+          <li class="navigation-item">
+            <!-- Сомневаюсь, что это ссылка для входа на сайт,
+              тк в макете называется Button -->
```
ориентироваться на то, что в макете написано про элемент, стоит с большой осторожностью. не все дизайнеры называют слои как-то осознанно, но даже те, кто к этому относится серьёзно, скорее всего будут ориентироваться на визуальную сторону, а не на техническую реализацию. то есть, если в макете написано, что это button, это с большей вероятностью означает, что этот элемент **выглядит** как кнопка. и с очень небольшой вероятностью означает, что этот элемент нужно делать тегом button. 

у нас нет непосредственных данных о том, что этот элемент делает. но если ссылка нативно умеет перевести нас на другую страницу (или к якорю на этой же странице), то кнопка (если она не является кнопкой отправки формы, а эта кнопка ей точно не является) по умолчанию не умеет делать ничего. предполагается, что для неё напишут какой-то скрипт, который добавит ей логику. поэтому в случае, когда неизвестно или трудно предположить, что делает элемент, его можно сделать ссылкой. по умолчанию у ссылки функциональности чуть больше

talequale commented on index.html:
```diff
@@ -5,8 +5,151 @@
+            <!-- Сомневаюсь, что это ссылка для входа на сайт,
+              тк в макете называется Button -->
+            <a class="navigation-link button" href="#">
+              <span class="visually-hidden">Вход</span>Хочу сюда!
```
тут есть какой-то текст, и он что-то означает. дублировать его скрытым текстом нет необходимости, во всяком случае до тех пор, пока мы точно не знаем, какая у элемента функция. а мы точно не знаем. я не уверена, что это кнопка для авторизации. возможно она открывает всё ту же форму поиска гостиницы, но это только предположение

talequale commented on index.html:
```diff
@@ -5,8 +5,151 @@
+         <p class="tagline">
+           <img class="tagline-image" src="#"  alt="Welcome to the georgeous SEDONA because the Grand Canyon sucks ">
+           <strong>Седона — небольшой городок в Аризоне,
+            заслуживающий большего!</strong>
```
этот элемент точно не относится к блоку с картинкой, он явно живёт в одном белом блоке вместе с текстовым параграфом про 5 причин 
![image](https://user-images.githubusercontent.com/11165551/169871223-6f26e194-e2b9-4d12-83a4-78c9f3ebe55f.png)

и как правило фразовые теги типа strong, b, em, i не живут сами по себе, по своему прямому назначению они выделяют куски текста в рамках параграфа. а конкретно этот текст в принципе можно сделать и заголовком, ничего страшного в этом не будет

talequale commented on index.html:
```diff
@@ -5,8 +5,151 @@
+         <h2 class="visually-hidden">Приветствие</h2>
+         <!-- нужно ли img в параграф с описанием? -->
+         <p class="tagline">
+           <img class="tagline-image" src="#"  alt="Welcome to the georgeous SEDONA because the Grand Canyon sucks ">
```
можно положить картинку в параграф, можно в div, можно вообще без обёртки обойтись. в блоке с фоном из гор у нас из контента есть только картинка и скрытый заголовок, что-то разделять необходимости нет

talequale commented on index.html:
```diff
@@ -5,8 +5,151 @@
+            заслуживающий большего!</strong>
+         </p>
+         <p class="tagline-discription">Рассмотрим 5 причин, по которым Седона круче, чем Гранд-Каньон!</p>
+       </section>
```
вряд ли эти два текстовых элементов относятся к секции с картинкой про седону. судя по тексту ("Рассмотрим 5 причин") они явно про причины, то есть относятся к секции advantages, туда бы их и запихнуть

talequale commented on index.html:
```diff
@@ -5,8 +5,151 @@
+       </section>
+       <section class="advantages">
+         <h2 class="visually-hidden">Преимущества</h2>
+         <ul class="advantages-list">
```
у нас нумерованный список из 5 причин, поэтому тут не ul (unordered list), а ol (ordered list) с пятью пунктами

talequale commented on index.html:
```diff
@@ -5,8 +5,151 @@
+             <h3 class="advantages-title">Настоящий городок</h3>
+             <!-- сомневаюсь, но выделила номера преимуществ в описание, тк считаю, что список 
+              неупорядоченный -->
+             <p class="advantage-number">- №1 -</p>
```
номера можно не добавлять в разметку, их можно будет сделать стилями (автоматическую нумерацию элементов в списке)

talequale commented on index.html:
```diff
@@ -5,8 +5,151 @@
+             <!-- сомневаюсь, но выделила номера преимуществ в описание, тк считаю, что список 
+              неупорядоченный -->
+             <p class="advantage-number">- №1 -</p>
+             <p class="advantages-discrition">Седона — не аттракцион для туристов, там течёт своя жизнь</p>
```
слова discrition не существует, есть description

talequale commented on index.html:
```diff
@@ -5,8 +5,151 @@
+             <p class="advantage-number">- №1 -</p>
+             <p class="advantages-discrition">Седона — не аттракцион для туристов, там течёт своя жизнь</p>
+             <img class="sedona-image" src="#" alt="Американский городок, мотель с флагами и видом на горы">
+             <!-- сделала список в списке, но не уверена, да они относятся к преимуществам городка, 
+              и их заголовки должны быть на уровень ниже или нет? -->
+             <ul class="advantages-list">
```
список в списке это правильно, этот список с иконками поясняет первый пункт про "Настоящий городок" (в том смысле, что признаки настоящего городка это какие-то особые "жильё", "еда" и "сувениры")

правильно, что этот вложенный список сделан тегом ul, т к это неупорядоченный список

неправильно сделала, что у этого вложенного списка тот же класс, что и у внешнего, потому что одинаковые классы = одинаковые стили, а стили у этих списков разные

OlgaNikl replied:
остался вопрос, у вложенного списка заголовки должны быть на уровень ниже <h4></h4> или 
также <h3></h3>, как главного списка?

talequale commented on index.html:
```diff
@@ -5,8 +5,151 @@
+             <ul class="advantages-list">
+              <li class="advantages-item">
+                <h3 class="advantages-title">Жильё</h3>
+                <p class="advantages-discrition">Рекомендуем пожить в настоящем мотеле, всё как в кино!</p>
```
"discrition" надо исправить во всём документе

talequale commented on index.html:
```diff
@@ -5,8 +5,151 @@
+            <p class="advantage-number">- №2 -</p>
+            <img class="bridge-image" src="" alt="Скалы в виде моста над пропастью">
+            <p class="advantages-discrition">Да, по нему можно пройти! Если вы осмелитесь, разумеется</p>
+           </li>
```
нужно поправить оформление во всём документе, сейчас в отступах разное количество пробелов, должно быть везде 2 
![Разметка всех страниц by OlgaNikl · Pull Request #2 · htmlacademy-htmlcss:2067969-sedona-34 2022-05-23 20-35-53](https://user-images.githubusercontent.com/11165551/169876133-81d700ab-19bf-4182-aefe-972d65907f07.jpg)

talequale commented on index.html:
```diff
@@ -5,8 +5,151 @@
+       </section>
+      </main>
+      <footer class="page-footer">
+        <!-- включила секцию о подписке в футер, тк инфо повторяется на страницах, но в макете этот не включен в футер -->
```
не всё, что повторяется, обязательно входит в шапку или в подвал. могут и другие блоки повторяться на страницах, это очень частый случай
в целом ничто не говорит о том, что блок подписки относится к футеру, поэтому я бы его в футер включать не стала

talequale commented on index.html:
```diff
@@ -5,8 +5,151 @@
+        <section class="footer-subscribe">
+          <h2 class="subscribe">Подпишитесь на рассылку</h2>
+          <p class="subscribe-discription">Только полезная информация и никакого спама, честное бойскаутское!</p>
+          <form class="subscribe-form" action="http:/echo.htmlacademy.ru/" method="post">
+              <label for="e-mail" class="visually-hidden">E-mail</label>
+              <input type="email" name="e-mail" id="e-mail" placeholder="Ваш e-mail">
+              <button type="submit" class="subscribe-button">Подписаться</button>
+          </form>
```
снова обращаю внимание на оформление, в этом блоке в отступе похоже по 4 пробела
![Разметка всех страниц by OlgaNikl · Pull Request #2 · htmlacademy-htmlcss:2067969-sedona-34 2022-05-23 20-39-35](https://user-images.githubusercontent.com/11165551/169876586-72b66d71-ed4f-48a0-b134-0213fadee13a.jpg)

такое бывает, если копировать куски кода из документа с другим форматированием. нужно либо руками поправить отступы, либо автоформатом

talequale commented on index.html:
```diff
@@ -5,8 +5,151 @@
+            <a class="contacts-phone" href="tel:88128121212">+7 (812) 812-12-12</a>
+        </address>
+       </section>
+       <section class="logohtmlacademy">
```
плохой класс, особенно учитывая, что его всего один символ отличает от класса ссылки. можно более абстрактный класс подобрать (copyright, ссылка - copyright-link, картинка - copyright-logo)

talequale reviewed (commented):
технический комментарий по коммитам:
верно, что системы контроля версий существуют для удобства разработчика, чтобы ничего не потерялось и чтобы можно было в любой момент откатиться к какому-то промежуточному состоянию. но именно ради того, чтобы можно было легко это делать, есть смысл придерживаться какой-то системы в том, как ты делаешь коммиты. 4 коммита с одинаковым коммит-месседжем на один только хедер страницы это многовато. если тебе нужно будет откатиться к определённому состоянию, сходу ты, конечно, не вспомнишь, в каком коммите ты меняла то, что тебе нужно (или наоборот, больше не нужно). делать по коммиту на каждую маленькую фичу (например, список навигации и т п) можно, но лучше описывать в коммит-месседже, что в этом коммите было сделано ("разметила список навигации")
то есть, грубо говоря, в процессе работы старайся делать коммит, когда какая-то небольшая задача закончена. и описывай эту задачу в сообщении коммита. дробить задачи очень хорошо, но во всём нужна мера

с разметкой всё в целом хорошо, сделано очень аккуратно. есть небольшие нюансы по семантике и оформлению кода

OlgaNikl committed:
Update index.html

изменила классы и текст подписи для пользовательского списка навигации

OlgaNikl committed:
Update index.html

изменила класс картинки в блоке приветствие; перенесла текст с описание о Седоне в блок с преимуществами; изменила список преимуществ на нумерованный; удалила нумерацию из разметки; исправила discription; исправила класс у вложенного списка (для др стилей); поправила отступы вручную

OlgaNikl committed:
Update index.html

переместила секцию Подписка в main, изменила классы; изменила класс для логотипа академии; заменила тег для Заинтересовались на параграф; поправила отступы еще раз;

OlgaNikl committed:
Create catalog.html

Создала новый файл Каталог, разметила header

OlgaNikl committed:
Update catalog.html

Разметила main, в нем header-main, включила в него форму с фильтрами для выбора гостиниц.

OlgaNikl committed:
Update catalog.html

переместила блок с фильтрами в отдельную секцию;  добавила часть вид и сортировка;

OlgaNikl committed:
Update catalog.html

добавила список гостиниц

OlgaNikl committed:
Update catalog.html

Добавила пагинацию; изменила заголовок секции и поправила рамки секции;

OlgaNikl committed:
Update catalog.html

заменила кнопку на ссылку (В ИЗБРАННОЕ); добавила секцию  подписка; добавила футер;

OlgaNikl committed:
Update catalog.html

добавила комментарий

OlgaNikl committed:
Update index.html

добавила ссылку на страницу каталога гостиниц

talequale commented on catalog.html:
```diff
@@ -0,0 +1,224 @@
+          </li>
+          <li class="user-list-item">
+            <a class="user-list-link" href="#">иконка сердечка</a>
+          </li>
```
я для примера написала "иконка лупы" и "иконка сердечка", полностью копировать не надо ) понятно, что это про поиск и избранное, описания нужны соответствующие

talequale commented on index.html:
```diff
@@ -5,8 +5,130 @@
+              <h3 class="advantages-title">Настоящий городок</h3>
+              <p class="advantages-description">Седона — не аттракцион для туристов, там течёт своя жизнь</p>
+              <img class="sedona-image" src="#" alt="Американский городок, мотель с флагами и видом на горы">
+              <!-- заголовки во вложенном списке должны быть на уровень ниже или нет? -->
```
да, на уровень ниже. иерархия заголовков именно так и работает

talequale commented on index.html:
```diff
@@ -5,8 +5,130 @@
+            <li class="advantages-item">
+              <h3 class="advantages-title">Там есть мост дьявола</h3>
+              <img class="bridge-image" src="" alt="Скалы в виде моста над пропастью">
+              <p class="advantages-description">Да, по нему можно пройти! Если вы осмелитесь, разумеется</p>
```
последовательность блоков непонятна, что нам даёт картинка между текстовыми элементами? по логике и по семантике без особой разницы, что раньше - картинка или текстовое описание, а строить сетку будет очень больно с картинкой на средней позиции

talequale commented on catalog.html:
```diff
@@ -0,0 +1,224 @@
+          </fieldset>
+          <button class="button" type="submit" value="Применить">Применить</button>
+          <button class="reset-button" type="reset" value="Сбросить">Сбросить</button>
+          <!-- не уверена, что верно разместила output -->
```
это две разные формы, для начала. форму  с фильтрами лучше вообще перенести в inner-header, хотя бы потому, что нам нужен блок, которому мы в стилях напишем фон с горами. иначе непонятно, как всё это рисовать, визуально всё, что в рамках блока с фоновой картинкой, это отдельный блок. а структурно у тебя получается, что нет. это сложно будет реализовать 

output вообще не нужен, достаточно просто спана с цифрой

talequale commented on index.html:
```diff
@@ -5,8 +5,130 @@
+        <section class="subscribe">
+          <h2 class="subscribe-title">Подпишитесь на рассылку</h2>
+          <p class="subscribe-discription">Только полезная информация и никакого спама, честное бойскаутское!</p>
+          <form class="subscribe-form" action="http:/echo.htmlacademy.ru/" method="post">
```
ошибка в адресе, должно быть `"http://echo.htmlacademy.ru"`

talequale commented on catalog.html:
```diff
@@ -0,0 +1,224 @@
+              <input class="price-input" type="number" name="price" id="min-price" min="0" max="9000" step="1" value="0">
+              <span class="min-price-input">от</span>
+              <input class="price-input" type="number" name="price" id="max-price" min="0" max="9000" step="1" value="9000">
+              <span class="max-price-input">до</span>
```
"от" и "до" это подписи полей

OlgaNikl replied:
сделала подписи полей, но они все равно не в поле, а рядом размещены.

talequale replied:
подпись = label. размещать их рядом с полями нормально, достаточно привязать подпись к полю атрибутом for

talequale replied:
то, что визуально подпись как бы находится в поле, решается стилями (несколькими способами причём)

talequale commented on catalog.html:
```diff
@@ -0,0 +1,224 @@
+              <span class="min-price-input">от</span>
+              <input class="price-input" type="number" name="price" id="max-price" min="0" max="9000" step="1" value="9000">
+              <span class="max-price-input">до</span>
+              <!-- здесь должен быть range с двумя бегунками -->
```
не должен, не бывает нативного range с двумя бегунками. как правило верстают нужного вида блок на дивах, спанах и кнопках, и потом пишут логику на js

OlgaNikl replied:
как мне это осуществить сейчас? 
авторы сказали, что это мы изучим в 9-ой части.

talequale replied:
пока можно не делать, я просто прокомментировала range, думала, ты имеешь в виду `input type="range"`

talequale commented on catalog.html:
```diff
@@ -0,0 +1,224 @@
+       <a href="#" class="button button-light">иконка карточек</a>
+       <a href="#" class="button">иконка галереи</a>
+       <a href="#" class="button">иконка списка</a>
+       <!-- нужно ли оборачивать список в section? -->
```
сам список не обязательно, можно блок с сортировкой и списком обернуть. и саму строку с сортировкой (количество результатов поиска, селект, кнопки сортировки) хорошо бы обернуть дивом, чтобы изолировать от списка логически и физически (так проще будет строить сетку)

talequale commented on catalog.html:
```diff
@@ -0,0 +1,224 @@
+           <a class="hotel-card-discription button" href="#">ПОДРОБНЕЕ</a>
+           <a class="hotel-card-favorites" href="#">В ИЗБРАННОЕ</a>
+           <img src="#" alt="4 звезды">
+           <a class="hotel-raiting" href="#">Рейтинг 8,5</a>
```
это вряд ли ссылка, просто информационный текст

talequale commented on catalog.html:
```diff
@@ -0,0 +1,224 @@
+           <a class="pagination-link">10</a>
+         </li>
+       </ul>
+       <!-- гостиниц на странице это вывод числ, это output? -->
+       <div class="item-amount">Гостиниц на странице:
+         <output name="amount">[4]</output>
+       </div>
```
нет, это выпадающий список, в котором пользователь может выбрать, сколько результатов поиска он хочет видеть на одной странице

talequale reviewed (commented):
проверь форматирование (отступы), на обеих страницах есть с этим проблемы

OlgaNikl committed:
Update index.html

для файла index: отформатировала отступы; заменила названия ссылок в навигации; переместила изображение над описанием в писке преимуществ; изменила уровень заголовка во вложенном списке преимуществ; исправила ошибку в адресе в форме подписки;

OlgaNikl committed:
Update catalog.html

переместила форму с фильтрами гостиниц в хэдер-иннер; изменила заголовок секции на Фильтр гостиниц; исправила ошибку в адресе формы подписки; добавила подписи полей к ценовому фильтру;  обернула в див сортировку, настройку вида и кол-во гостиниц; изменила ссылку Рейтинг на параграф;
